plasmapy/dispersion/two_fluid_dispersion.py

```diff
@@ -0,0 +1,352 @@
+        If applicable arguments are not instances of `~astropy.units.Quantity` or
+        cannot be converted into one.
+
+    TypeError
+        If ``ion`` is not of type or convertible to `~plasmapy.particles.Particle`.
+
+    TypeError
+        If ``gamma_e``, ``gamma_i``, or``z_mean`` are not of type `int` or `float`.
+
+    ~astropy.units.UnitTypeError
+        If applicable arguments do not have units convertible to the expected
+        units.
+
+    ValueError
+        If the ``B``, ``k``, ``n_i``, ``T_e``, or ``T_i`` is negative.
+
+    ValueError
+        If ``k`` is negative or zero.
+
+    ValueError
+        If ``ion`` is not of category ion or element.
+
+    ValueError
+        If ``B``, ``n_i``, ``T_e``, or ``T_I`` are not single valued
+        `astropy.units.Quantity` (i.e. an array).
+
+    ValueError
+        If ``k`` or ``theta`` are not single valued or a 1-D array.
+
+    Warns
+    -----
+    : `~plasmapy.utils.exceptions.PhysicsWarning`
+        When the computed wave frequencies violate the low-frequency
+        (:math:`\omega/kc \ll 1`) assumption of the dispersion relation.
+
+    Notes
+    -----
+
+    The complete dispersion equation presented by Springer 1963 [2]_ (equation 1
+    of Bellan 2012 [1]_) is:
+
+    .. math::
+        \left( \cos^2 \theta - Q \frac{\omega^2}{k^2 {v_A}^2} \right) &
+        \left[
+            \left( \cos^2 \theta - \frac{\omega^2}{k^2 {c_s}^2} \right)
+            - Q \frac{\omega^2}{k^2 {v_A}^2} \left(
+                1 - \frac{\omega^2}{k^2 {c_s}^2}
+            \right)
+        \right] \\
+            &= \left(1 - \frac{\omega^2}{k^2 {c_s}^2} \right)
+              \frac{\omega^2}{{\omega_{ci}}^2} \cos^2 \theta
+
+    where
+
+    .. math::
+        Q &= 1 + k^2 c^2/{\omega_{pe}}^2 \\
+        \cos \theta &= \frac{k_z}{k} \\
+        \mathbf{B_o} &= B_{o} \mathbf{\hat{z}}
+
+    :math:`\omega` is the wave frequency, :math:`k` is the wavenumber, :math:`v_A`
+    is the Alfvén velocity, :math:`c_s` is the sound speed, :math:`\omega_{ci}` is
+    the ion gyrofrequency, and :math:`\omega_{pe}` is the electron plasma frequency.
+    This relation does additionally assumme low-frequency waves
+    :math:`\omega/kc \ll 1`, no D.C. electric field :math:`\mathbf{E_o}=0` and
+    quasi-neutrality.
+
+    Following section 5 of Bellan 2012 [1]_ the exact roots of the above dispersion
+    equation can be derived and expressed as one analytical solution (equation 38
+    of Bellan 2012 [1]_):
+
+    .. math::
+        \frac{\omega}{\omega_{ci}} = \sqrt{
+            2 \Lambda \sqrt{-\frac{P}{3}} \cos\left(
+                \frac{1}{3} \cos^{-1}\left(
+                    \frac{3q}{2p} \sqrt{-\frac{3}{p}}
+                \right)
+                - \frac{2 \pi}{3}j
+            \right)
+            + \frac{\Lambda A}{3}
+        }
+
+    where :math:`j = 0` represents the fast mode, :math:`j = 1` represents the
+    Alfvén mode, and :math:`j = 2` represents the Acoustic mode.  Additionally,
```

Comment:
```suggestion
    Alfvén mode, and :math:`j = 2` represents the acoustic mode.  Additionally,
```
I usually see acoustic/magnetoacoustic/magnetosonic lower case, though as long as it's consistent it would be fine with me.